File: frontend/src/types/smart-apply.ts
import type { Application } from "@/types/application";
import type { Email } from "@/types/email";

export interface SmartApplyRequest {
  raw_text: string;
}

export interface SmartApplyMeta {
  extraction_confidence: string;
}

export interface SmartApplyResponse {
  application: Application;
  email: Email;
  meta: SmartApplyMeta;
}

File: api-gateway/internal/model/smart_apply.go
package model

type SmartApplyRequest struct {
	RawText string `json:"raw_text"`
}

type SmartApplyMeta struct {
	ExtractionConfidence string `json:"extraction_confidence"`
}

type SmartApplyResponse struct {
	Application *Application   `json:"application"`
	Email       *Email         `json:"email"`
	Meta        SmartApplyMeta `json:"meta"`
}
